spec/defines/host_spec.rb

```diff
@@ -20,7 +20,34 @@
         end
 
         it {
-          verify_concat_fragment_exact_contents(catalogue, 'dhcp.hosts+10_myhost.hosts', [
+          verify_concat_fragment_with_comments(catalogue, 'dhcp.hosts+10_myhost.hosts', [
+            'host myhost {',
+            '  hardware ethernet   01:02:03:04:05:06;',
+            '  fixed-address       10.0.0.100;',
+            '  ddns-hostname       "myhost";',
+            '}',
+          ])
+        }
+      end
+
+      describe 'comment parameter' do
+        let :params do {
+          :ip      => '10.0.0.100',
+          :mac     => '01:02:03:04:05:06',
+          :comment => 'a useful comment',
+        } end
+
+        let :facts do
+          facts
+        end
+
+        let :pre_condition do
+          "class { '::dhcp': interfaces => ['eth0']}"
+        end
+
+        it {
+          verify_concat_fragment_with_comments(catalogue, 'dhcp.hosts+10_myhost.hosts', [
```

Comment:
Can't you just use this?
```ruby
let(:expected_content) do
  <<~CONTENT
...
  CONTENT
end

it {  is_expected.to contain_concat__fragment('dhcp.hosts+10_myhost.hosts').with_content(expected_coontent) }
```

Reply:
I don't see why not.  Do you want me to clean it up?

Reply:
Please do, then we can merge and include this in the release being put together for this week.

Reply:
Copy. I was only trying to match the style of the existing code.  Will clean it up shortly.

Reply:
I'll be honest and say that a lot is rather old and may have been written with incomplete understanding of rspec or that it simply wasn't that powerful back then.

Reply:
I've opened #195 to remove the concat helper methods.  Dealing with the whitespace without add new magic helper methods was miserable.

Reply:
I'm planning to rebase and rework this PR if/when #195 is merged.